lib/seccomp-tools/cli/cli.rb

```diff
@@ -1,21 +1,30 @@
 # frozen_string_literal: true
 
+require 'os'
 require 'seccomp-tools/cli/asm'
 require 'seccomp-tools/cli/disasm'
-require 'seccomp-tools/cli/dump'
+require 'seccomp-tools/cli/dump' if OS.linux?
 require 'seccomp-tools/cli/emu'
 require 'seccomp-tools/version'
 
 module SeccompTools
   # Handle CLI arguments parse.
   module CLI
     # Handled commands
-    COMMANDS = {
-      'asm' => SeccompTools::CLI::Asm,
-      'disasm' => SeccompTools::CLI::Disasm,
-      'dump' => SeccompTools::CLI::Dump,
-      'emu' => SeccompTools::CLI::Emu
-    }.freeze
+    COMMANDS = if OS.linux?
+                 {
+                   'asm' => SeccompTools::CLI::Asm,
+                   'disasm' => SeccompTools::CLI::Disasm,
+                   'dump' => SeccompTools::CLI::Dump,
+                   'emu' => SeccompTools::CLI::Emu
+                 }.freeze
+               else
+                 {
+                   'asm' => SeccompTools::CLI::Asm,
+                   'disasm' => SeccompTools::CLI::Disasm,
+                   'emu' => SeccompTools::CLI::Emu
+                 }.freeze
+               end
```

Comment:
I would prefer doing in this way:

Let `COMMANDS` be all the commands (including dump, i.e. revert changes of this file), and modify `lib/seccomp-tools/cli/dump.rb` and `lib/seccomp-tools/dumper.rb` instead.

For `lib/seccomp-tools/dumper.rb`, add a constant under `module Dumper`, maybe named `ENABLED`, with value `OS.linux?`. Also add the condition to require ptrace only when `OS.linux?` like you did here.
Also in dumper.rb, add `return [] unless ENABLED` for both `dump` and `dump_by_pid` methods.

For `lib/seccomp-tools/cli/dump.rb`, modify `USAGE` to add something like "NOTE: this function is only available on Linux."
Also in `cli/dump.rb`, add `return warn(Util.colorize("Dump is only available on Linux.", t: :error)) unless Dumper::ENABLED` (and you will need add `require 'seccomp-tools/util'` for the utility module)